### PR TITLE
Fix/logging macro redefinition in demos

### PR DESCRIPTION
**Pull request by aggarw13**
Fix macro redefinition error for `LIBRARY_LOG_NAME` and `LIBRARY_LOG_LEVEL` in MQTT and HTTP demos. 

*Description of changes:*
Fix includes defining the logging macros within `#ifndef` blocks in the config files, and including `demo_config.h` before any headers in demo source files.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/http/http_demo_plaintext/demo_config.h [hunk 1]
```diff
@@ -26,19 +26,23 @@
 /******* DO NOT CHANGE the following order ********/
 /**************************************************/
 
-/* Logging related header files are required to be included in the following order:
+/* Logging config definition and header files inclusion are required in the following order:
```
Comment (gkwicker) [inline demos/http/http_demo_plaintext/demo_config.h:29]: Reword: _"Include logging header files and define logging macros in the following order."_

File: demos/http/http_demo_plaintext/http_config.h [hunk 2]
```diff
@@ -26,21 +26,27 @@
 /******* DO NOT CHANGE the following order ********/
 /**************************************************/
 
-/* Logging related header files are required to be included in the following order:
+/* Logging config definition and header files inclusion are required in the following order:
```
Comment (gkwicker) [inline demos/http/http_demo_plaintext/http_config.h:29]: Same comment as previous file, line 29 – reword throughout this PR.

File: demos/http/http_demo_plaintext/demo_config.h (hunk 1 above)
Comment (aggarw13) [reply demos/http/http_demo_plaintext/demo_config.h]: Replace the line across all config files.

**Pull request merged**